app/src/main/java/com/ggetters/app/data/repository/match/CombinedMatchEventRepository.kt

```
@@ -18,108 +24,194 @@ class CombinedMatchEventRepository @Inject constructor(
+                    )
+                }
+            }
+        }
+
+        awaitClose {
+            offlineJob.cancel()
+            remoteJob.cancel()
+        }
+    }.distinctUntilChanged()
+
+    override fun getEventsByMatchIdAndType(matchId: String, eventType: String): Flow<List<MatchEvent>> = channelFlow {
```

detekt (1.23.8)

[warning] The function getEventsByMatchIdAndType appears to be too complex based on Cognitive Complexity (complexity: 27). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)